test/helpers/position_assertions.cc

```diff
@@ -546,6 +571,128 @@ void UsageAssertion::check(const UnorderedMap<string, shared_ptr<core::File>> &s
     }
 }
 
+void HighlightAssertion::check(const UnorderedMap<string, shared_ptr<core::File>> &sourceFileContents,
+                               LSPWrapper &lspWrapper, int &nextId, string_view uriPrefix, string_view symbol,
+                               const Location &queryLoc, const vector<shared_ptr<RangeAssertion>> &allLocs) {
+    const int line = queryLoc.range->start->line;
+    // Can only query with one character, so just use the first one.
+    const int character = queryLoc.range->start->character;
+    auto uri = queryLoc.uri;
+    auto locSourceLine = getLine(sourceFileContents, uriPrefix, queryLoc);
+    string locFilename = uriToFilePath(uriPrefix, queryLoc.uri);
+
+    auto referenceParams =
+        make_unique<ReferenceParams>(make_unique<TextDocumentIdentifier>(queryLoc.uri),
+                                     // TODO: Try with this false, too.
+                                     make_unique<Position>(line, character), make_unique<ReferenceContext>(true));
+    int id = nextId++;
+    auto responses = lspWrapper.getLSPResponsesFor(*makeDocumentHighlightRequest(id, queryLoc.uri, line, character));
+    if (responses.size() != 1) {
+        EXPECT_EQ(1, responses.size())
+            << "Unexpected number of responses to a `textDocument/documentHighlight` request.";
+        return;
+    }
+
+    if (assertResponseMessage(id, *responses.at(0))) {
+        auto &respMsg = responses.at(0)->asResponse();
```

Comment:
Could we move the shared logic into a helper method or maybe a shared parent class b/w UsageAssertion and HighlightAssertion? The basic premise is identical:

* Make query at a specific position.
* Check that the resulting locations match a set of expected locations.

I know that `DocumentHighlight` returns highlights that are not `Location`s, but you could convert them into `Location`s trivially and define a common interface. That'll also prevent the need for a separate `getLine`.